Hardcore.lua

```diff
@@ -144,6 +144,56 @@ local ALERT_STYLES = {
 -- the big frame object for our addon
 local Hardcore = CreateFrame("Frame", "Hardcore", nil, "BackdropTemplate")
 Hardcore.ALERT_STYLES = ALERT_STYLES
+Hardcore.verify_shown = false
```

Comment:
This state feels weird. Is there a specific reason you did not use the Show and Hide functions of the frame itself?

```
widget:Show()
widget:Hide()
```

Reply:
> This state feels weird. Is there a specific reason you did not use the Show and Hide functions of the frame itself?
> 
> ```
> widget:Show()
> widget:Hide()
> ```

Other than being being a noob? None that come to mind - If you have a quick-fix then be my guest!

My intent was to ensure that there would only ever be one Ace3 Window Dialog available (for this feature) at once. I missed the mark. I suspect there's a problem with multiple windows if the slash command is used while a window is already showing - I haven't tested it.

"Get it perfect now", "Get it out the door now, fix it later" ?

Reply:
> This state feels weird. Is there a specific reason you did not use the Show and Hide functions of the frame itself?

Turns out, there is:
```lua
function Hardcore:CloseVerifiedDialog(widget)
	Hardcore.verify_shown = false
	AceGUI:Release(widget)
end```

Ace3 is recycling the dialog's resources for use elsewhere when it's not shown.. it's not 'hiding it', it's destroying it.

Reply:
You are creating a new widget every time call `Hardcore:GetVerifiedDialog` which I assume is the problem.

You could make the dialog a singleton and use :Hide() and :Show() to toggle its visibility

I haven't tested if the code works, but something like:
```
GetVerifiedDialog = nil -- this should be in the global scope

function Hardcore:CreateGetVerifiedDialog() -- call this function once when the addon is initialized so the dialog is ready when needed
 -- create widget with ace 3 in here
end
```

And then toggle the dialog whenever needed, e.g. when clicking the Get Verified button defined in the xml file:

```
<OnClick>
    GetVerifiedDialog:Show()
</OnClick>
```

Reply:
> You are creating a new widget every time call `Hardcore:GetVerifiedDialog` which I assume is the problem.
> 
> You could make the dialog a singleton and use :Hide() and :Show() to toggle its visibility

That would be the pattern in GUIs other than Ace3, however in WoW you can't ever destroy something - once created, it's always there until the GC is forced to handle it.

Ace3 overcomes this by using a fixed set of resources, and restyling/repositioning them for each different dialog you view: the heading frame and close box on the String Exporter is the same as used in another Ace3 addon's UI - there's 'enough' if you want to run multiple Ace3 apps and have them displayed at the same time, but there's still only a finite number of them.

The pattern used above allows Ace3, once the window is 'hidden' to reallocate the resources for use on other dialogs. Our dialog ceases to exist - it's not actually 'hidden', but destroyed.

Thus, when trying to show multiple instances of this window, we need to avoid RE-creating it, if it's already there and showing - that would work, but it would occupy more 'can never be reclaimed' slots (one per widget component)

So, yes - the pattern is strange, but I understand it - the code as committed is correct.

Reply:
> as it's not 'Show/Hide' but 'Build/Destroy'

Isn't that the same thing?
Build the widget when needed (e.g. when clicking the Get Verified button)
Destroy the widget when no longer needed (when closing the window)

Either way there shouldn't be the need to have external state to display/hide (create/destroy) a widget.